sched/sched/sched.h

```diff
@@ -320,7 +320,9 @@ int nxthread_create(FAR const char *name, uint8_t ttype, int priority,
 /* Task list manipulation functions */
 
 bool nxsched_add_readytorun(FAR struct tcb_s *rtrtcb);
-bool nxsched_remove_readytorun(FAR struct tcb_s *rtrtcb, bool merge);
+bool nxsched_remove(FAR struct tcb_s *rtrtcb);
+void nxsched_remove_running(FAR struct tcb_s *rtrtcb);
+void nxsched_remove_not_running(FAR struct tcb_s *rtrtcb);
```

Comment:
Why is such odd interface being exposed? `nxsched_remove()` ? What exactly is being removed?
I reject the inclusion of such PRs into the community, as they merely optimize branch cases, which not only reduce the readability of the code and also make its maintenance more difficult.